src/python/pants/option/arg_splitter.py

```diff
@@ -113,26 +97,16 @@ def __init__(self, known_scope_infos: Iterable[ScopeInfo], buildroot: str) -> No
         # List of pairs (prefix, ScopeInfo).
         self._known_scoping_prefixes = [(f"{si.scope}-", si) for si in sorted_scope_infos]
 
-    @property
-    def help_request(self) -> HelpRequest | None:
-        return self._help_request
-
-    def _check_for_help_request(self, arg: str) -> bool:
-        if arg not in self._HELP_ARGS:
-            return False
-        if arg in self._HELP_VERSION_ARGS:
-            self._help_request = VersionHelp()
-        elif arg in self._HELP_ALL_SCOPES_ARGS:
-            self._help_request = AllHelp()
-        else:
-            # First ensure that we have a basic OptionsHelp.
-            if not self._help_request:
-                self._help_request = ThingHelp()
-            # Now see if we need to enhance it.
-            if isinstance(self._help_request, ThingHelp):
-                advanced = self._help_request.advanced or arg in self._HELP_ADVANCED_ARGS
-                self._help_request = dataclasses.replace(self._help_request, advanced=advanced)
-        return True
+    @staticmethod
+    def _get_known_goal_scopes(
+        known_scope_infos: Iterable[ScopeInfo],
+    ) -> Iterator[tuple[str, ScopeInfo]]:
+        for si in known_scope_infos:
+            if not si.is_goal:
+                continue
+            yield si.scope, si
+            for alias in si.scope_aliases:
+                yield alias, si
```

Comment:
This method replaces all the variations we got rid of from the bespoke `_HELP_ARGS` tuple, using the new scope aliases support.